### PR TITLE
Migrate "beam_PerformanceTests_AvroIOIT" Jenkins jobs to GitHub Actions

### PR DESCRIPTION
**Pull request by andreydevyatkin**
The following jenkins jobs have been migrated:
- https://ci-beam.apache.org/job/beam_PerformanceTests_AvroIOIT/
- https://ci-beam.apache.org/job/beam_PerformanceTests_AvroIOIT_HDFS/

Added the `test-arguments-action` action to solve the following points:
- Encapsulate common test arguments into one place
- Apply arguments preparation logic (quotas, commas, etc.) to have a single template for creating config files for performance and load tests
- Allows to specify arguments that are generated at runtime.

### How has it been tested?

The jobs have been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by manually running the job (`workflow_dispatch`)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by andreydevyatkin** [comment]
The `.github/actions/setup-k8s-access/action.yml` action will disappear from changes once #28530 is merged

File: .github/workflows/beam_PerformanceTests_AvroIOIT.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Performance Tests AvroIOIT
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 0,12 * * *'
```
Comment (Abacn) [inline .github/workflows/beam_PerformanceTests_AvroIOIT.yml:22]: until github action supports random schedule let's add some manual randomness to performance and load test workflow - in order to distribute the stress

File: .github/workflows/performance-tests-job-configs/config_AvroIOIT.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+"--tempRoot=gs://temp-storage-for-perf-tests",
+"--bigQueryDataset=beam_performance",
+"--bigQueryTable=avroioit_results",
+"--influxMeasurement=avroioit_results",
+"--numberOfRecords=225000000",
+"--expectedHash=2f9f5ca33ea464b25109c0297eb6aecb",
+"--datasetSize=1089730000",
+"--numWorkers=5",
+"--autoscalingAlgorithm=NONE",
+"--influxDatabase=beam_test_metrics",
```
Comment (Abacn) [inline .github/workflows/performance-tests-job-configs/config_AvroIOIT.txt:29]: same as https://github.com/apache/beam/pull/28450#discussion_r1333171697 a load test builder step to set common options (e.g., currently if influxHost has changed then it needs to change all files)

File: .github/workflows/beam_PerformanceTests_AvroIOIT.yml (hunk 1 above)
Comment (andreydevyatkin) [reply .github/workflows/beam_PerformanceTests_AvroIOIT.yml]: updated

File: .github/workflows/performance-tests-job-configs/config_AvroIOIT.txt (hunk 2 above)
Comment (andreydevyatkin) [reply .github/workflows/performance-tests-job-configs/config_AvroIOIT.txt]: @Abacn I added the `test-arguments-action` action in order to encapsulate common test arguments into one place and get rid of different config file templates (quotes, commas, etc.). Unfortunately, there is no a ton of common arguments for both Performance and Load Tests, and moreover we have a different set of arguments with different case which depends on the language (java, python and go)

Comment (Abacn) [reply .github/workflows/performance-tests-job-configs/config_AvroIOIT.txt]: thanks! this is a good start. Other than the synthetic options, other options should share at least within a group and one can use action parameters. We can clean up as followup.

**Review by Abacn — APPROVED** [review]

**Comment by Abacn** [comment]
LGTM. since it is requisite for other tests let's merge this and see the regular run results.

**Pull request merged**